File: src/test/java/io/bloco/faker/components/AvatarTest.java
package io.bloco.faker.components;

import org.junit.Before;
import org.junit.Test;

import io.bloco.faker.Faker;

import static org.hamcrest.CoreMatchers.containsString;
import static org.hamcrest.CoreMatchers.not;
import static org.junit.Assert.assertThat;

public class AvatarTest {

    private Faker faker;

    @Before
    public void setUp() throws Exception {
        faker = new Faker();
    }

    @Test
    public void imageDefault() throws Exception {
        String defaultUrl = faker.avatar.image();
        assertThat(defaultUrl, containsString("robohash.org"));
        assertThat(defaultUrl, containsString("image"));
        assertThat(defaultUrl, containsString("300x300"));
        assertThat(defaultUrl, containsString("png"));
        assertThat(defaultUrl, containsString("set1"));
        assertThat(defaultUrl, not(containsString("bgset")));
    }

    @Test
    public void imageParams() throws Exception {
        String specificUrl = faker.avatar.image("my-slug", "80x80", "bmp", "set2", "bgset1");
        assertThat(specificUrl, containsString("robohash.org"));
        assertThat(specificUrl, containsString("my-slug"));
        assertThat(specificUrl, containsString("80x80"));
        assertThat(specificUrl, containsString("bmp"));
        assertThat(specificUrl, containsString("set2"));
        assertThat(specificUrl, containsString("bgset1"));
    }

    @Test(expected = IllegalArgumentException.class)
    public void imageInvalidSize() {
        faker.avatar.image("my-slug", "123456");
    }

    @Test(expected = IllegalArgumentException.class)
    public void imageInvalidFormat() {
        faker.avatar.image("my-slug", "80x80", "pdf");
    }
}

File: lib/src/test/kotlin/io/bloco/faker/components/AvatarTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import junit.framework.TestCase.assertFalse
import org.junit.Test
import kotlin.test.assertTrue

class AvatarTest {
    private val faker = Faker()

    @Test
    fun imageDefault() {
        val defaultUrl = faker.avatar.image()
        assertTrue(defaultUrl.contains("robohash.org"))
        assertTrue(defaultUrl.contains("image"))
        assertTrue(defaultUrl.contains("300x300"))
        assertTrue(defaultUrl.contains("png"))
        assertTrue(defaultUrl.contains("set1"))
        assertFalse(defaultUrl.contains("bgset"))
    }

    @Test
    fun imageParams() {
        val specificUrl = faker.avatar.image("my-slug", "80x80", "bmp", "set2", "bgset1")
        assertTrue(specificUrl.contains("robohash.org"))
        assertTrue(specificUrl.contains("my-slug"))
        assertTrue(specificUrl.contains("80x80"))
        assertTrue(specificUrl.contains("bmp"))
        assertTrue(specificUrl.contains("set2"))
        assertTrue(specificUrl.contains("bgset1"))
    }

    @Test(expected = IllegalArgumentException::class)
    fun imageInvalidSize() {
        faker.avatar.image("my-slug", "123456")
    }

    @Test(expected = IllegalArgumentException::class)
    fun imageInvalidFormat() {
        faker.avatar.image("my-slug", "80x80", "pdf")
    }
}